pkg/controllers/provisioning/v1alpha1/allocation/constraints.go

```diff
@@ -80,12 +81,11 @@ func (c *Constraints) getNodeOverhead(ctx context.Context, node *v1.Node) (v1.Re
 	}
 
 	// 2. filter DaemonSets to include those that will schedule on this node
-	podSpecs := []*v1.PodSpec{}
+	pods := []*v1.Pod{}
```

Comment:
Ah -- this is a good callout. I think customers should do this explicitly and taint their GPU workloads if they want this behavior.

Reply:
really? I think this should be a sane default that karpenter does for you.  I'm guessing that anyone who would want non-gpu pods to schedule on gpu instances would be a much smaller population than those who would want them separate.

Reply:
This is the kubernetes guidance for running GPU workloads. If we magically taint these nodes, then there's no mechanism to schedule CPU pods into the gaps on GPU nodes.

Reply:
(I'll preface this with: I don't know very much about GPU workloads on k8s and I'm okay with doing what you're suggesting)  
 
I would expect that the extra resources on a GPU machine would be reserved for GPU-only workloads just because of the cost of using up the other resources and not having space to schedule GPU workloads if that would happen later. A GPU frag'd cluster would be crazy expensive.